CMakeLists.txt

```diff
@@ -84,6 +84,10 @@ endif()
 # Plugins
 add_subdirectory(plugins)
 if (BUILD_FRONTEND)
+  if (ENABLE_GL)
+    target_link_libraries(plugins INTERFACE core_gl)
+    target_sources(plugins INTERFACE $<TARGET_OBJECTS:core_gl>)
```

Comment:
This feels conceptionally wrong. the core should not be a public depedency of the plugins. Individual plugins already links the core_gl (private). Also putting core objects into the plugins target is probably not the best place.